src/features/world/reducer.js

```diff
@@ -64,7 +64,7 @@ const worldReducer = (state = initialState, { type, payload }) => {
 
         case 'OPEN_CHEST':
             const { x, y } = payload;
-            const chest = state.chests[chestName(x, y)];
+            const chest = state.chests[chestName(state.currentMap, x, y)];
```

Comment:
Could you offer an explanation to why such a small change fixes the bug?

Reply:
The way Chests work is that they're a different level of tile (specifically, 4 for an open chest and -2 for an empty chest). When the player knocks into a tile with value 4, the next thing that happens is a call on line 62 in file src/features/player/actions/move-player.js, which calls through to exploreChest. This will dispatch the 'OPEN_CHEST' type (which is handled in the reducer file I changed, as you can tell).

Now, as you can imagine, this is called every time the player collides with the chest. Of course, on the first time around this is fine: We didn't have any item there anyways, so it'll be generated further down the line. So the question becomes -- why is it only on the third (and subsequent entries) that this becomes an issue, and not before hand? Well, that's more to do with the 'SET_CHEST_DATA' type. So, on the second time around, once 'OPEN_CHEST' is dispatched, it will set the item to null (after the second 'SET_CHEST_DATA' dispatch) and because from then on we simply read from the chest's current data in the state every time we go to dispatch the 'SET_CHEST_DATA' we'll only ever see null, because state.chests[chestName(x, y)] will always return undefined.

As to why it always returns undefined: At one point (specifically, when I fixed the last bug about duplicated chests), I changed the chestName function so that it returned a value in the form 'mapname(x, y)', however apparently the above line got lost in the changeover. In response to that, the above function call would always return undefined simply because we were calling chestName with the wrong arguments, and thus it always returned a key that wasn't found in the object, and so always gave us undefined. Which made the following if always execute.

So, by changing the call to be correct, it doesn't always return undefined (in fact, it should only do so if the specific chest isn't already 'opened') and won't always execute the inner block, thus meaning we should be able to always reopen the chest with the correct data (because we no longer overwrite it).